src/pages/api/modules/usenet/pause.ts

```diff
@@ -25,13 +26,43 @@ async function Post(req: NextApiRequest, res: NextApiResponse) {
       throw new Error(`Service with ID "${req.query.serviceId}" could not be found.`);
     }
 
-    if (!service.apiKey) {
-      throw new Error(`API Key for service "${service.name}" is missing`);
-    }
+    let result;
+    switch (service.type) {
+      case 'NZBGet': {
+        const url = new URL(service.url);
+        const options = {
+          host: url.hostname,
+          port: url.port,
+          login: service.username,
+          hash: service.password,
+        };
+
+        const nzbGet = NzbgetClient(options);
 
-    const { origin } = new URL(service.url);
+        result = await new Promise((resolve, reject) => {
+          nzbGet.pauseDownload(false, (err: any, result: any) => {
```

Comment:
Is there no definition available for ``err`` and ``result``?

Reply:
This is the callback that gets passed all the way to the underlying [`node-json-rpc`](https://www.npmjs.com/package/node-json-rpc) library. They don't provide any types for this callback in that library.

Reply:
No worries. We'll close this then